server/api/users.js

```diff
@@ -70,39 +71,43 @@ router.get("/:userId", requireToken, async (req, res, next) => {
  * the user's cart.
  * Returns the state of the user's cart.
  */
-router.put("/:userId/cart", requireToken, async (req, res, next) => {
-    if (!req.body) return res.status(304).send()
 
-    try {
-      const user = await User.findByPk(req.user.id, {
-        include: 'cart'
-      })
+router.put("/:userId/cart", requireToken, async (req, res, next) => {
 
-      const { cart } = user
-      
-      const productList = coalesceCart(req.body)
-      
-      // ought to be doing this in a transaction, but no time
+  try{
+    const cart = await Order.findByPk(req.user.cartId)
 
-      // clear out DB's copy of cart
-      OrderItem.destroy({
-        where: { orderId: cart.id }
-      })
+    const currentProd = await OrderItem.findOne({
+      where: {
+        [Sequelize.Op.and]: [
+          {orderId: cart.id},
+          {productId: req.body.productId}
+        ]
+      },
+    })
 
-      // add products into cart, setting quantity as we go
-      await OrderItem.bulkCreate(productList.map(p => ({
-        productId: p.id,
-        orderId: cart.id,
-        quantity: p.quantity
-      })))
+    await currentProd.update({productId: req.body.productId, quantity: req.body.quantity})
```

Comment:
what happens if no OrderItem is found?

Reply:
based on reducer logic, this shouldn't happen.  we can add more error handling in a different PR if we want